### ISSUE TITLE
Use of provider vs. RESTDataSource

### ISSUE THREAD

**Issue by nathanchapman** [Urigo/graphql-modules#174]
I would love to see some examples in the documentation of when to use a `provider` vs. `RESTDataSource`

Do they work together? It seems like a `provider` would be re-implementing a lot of methods that already exist in a `RESTDataSource`. Also I'm guessing any `RESTDataSource`s would go into the `contextBuilder`?

**Comment by ardatan** [comment]
DataSources looks quite similar to providers while providers feature is part of Dependency Injection and can be connected to each other in modular approach. Like schema and resolvers, providers can be considered as injectable data sources as in Apollo.

**Comment by dotansimha** [comment]
That's a great questions, @nathanchapman ! 
@ardatan is right, `DataSources` of apollo-server is similar to `Provider`s in graphql-modules.
Actually, before we started to develop graphql-modules, we used `DataSources` in some of our projects. 
`DataSources` is great as a simple bridge for your data, it's an abstraction that prevents from your resolvers to access directly the data source.
The problem is that each `DataSource` stands by itself, and they cannot interact with each other easily, and `Provider`s can interact and communicate with other `Provider`s with dependency-injection.
